### PR TITLE
Walrus operator support

### PR DESCRIPTION
**Pull request by BPHarris**
## What / How

* Treat walrus assignments (`ast.NamedExpr`) as normal assignments when the RHS does not include a lambda
* When the RHS contains a lambda appropriately peek inwards and copy the results into the left-hand side (i.e. `a = (b := lambda ...)` should process the lambda and assign it to `b` and `a`)

### CODE REVIEW

**Comment by BPHarris** [comment]
@olliemath This is ready for review :)

File: rattr/analyser/util.py [hunk 1]
```diff
@@ -662,6 +663,31 @@ def __exit__(self, *_):
         pass
 
 
+class changes:
```
Comment (olliemath) [inline rattr/analyser/util.py:666]: I would still make this a `Changes` class even if it's to be used as a context-manager

File: rattr/analyser/util.py [hunk 2]
```diff
@@ -662,6 +663,31 @@ def __exit__(self, *_):
...
+    """Context manager to return the dict-like's changes across the block."""
+
+    def __init__(self, target, cb=lambda t: t.keys()):
+        self.target = target
+        self.cb = cb
```
Comment (olliemath) [inline rattr/analyser/util.py:671]: `cb` is a bad name - what does it stand for?

**Comment by olliemath** [comment]
Hey @BPHarris is this still needing a merge? A few comments but otherwise looks good

**Comment by BPHarris** [comment]
> Hey @BPHarris is this still needing a merge? A few comments but otherwise looks good

Yeah, all comments addressed :)

**Review by olliemath — APPROVED** [review]

**Pull request merged**